websites/views.py

```diff
@@ -236,19 +240,60 @@ def publish(self, request, name=None):
         return self.publish_version(name, VERSION_LIVE, request)
 
     @action(
-        detail=True, methods=["post"], permission_classes=[HasWebsitePublishPermission]
+        detail=True,
+        methods=["post", "get"],
+        permission_classes=[HasWebsitePublishPermission],
     )
     def unpublish(self, request, name=None):
         """Unpublish the site and trigger the remove-unpublished-sites pipeline"""
         try:
             website = self.get_object()
-
-            Website.objects.filter(pk=website.pk).update(
-                unpublish_status=PUBLISH_STATUS_NOT_STARTED,
-                last_unpublished_by=request.user,
-            )
-            trigger_unpublished_removal(website)
-            return Response(status=200)
+            if request.method == "GET":
+                ocw_www_dependencies = WebsiteContent.objects.filter(
+                    (
+                        Q(type=CONTENT_TYPE_COURSE_LIST)
+                        | Q(type=CONTENT_TYPE_RESOURCE_COLLECTION)
+                    ),
+                    (
+                        Q(metadata__courses__icontains=website.name)
+                        | Q(metadata__resources__content__icontains=website.name)
+                    ),
+                    website__name=settings.ROOT_WEBSITE_NAME,
+                )
+                course_content_dependencies = WebsiteContent.objects.filter(
+                    ~Q(website__name=website.name),
+                    type=CONTENT_TYPE_PAGE,
+                    markdown__icontains=website.name,
+                )
+                course_dependencies = Website.objects.filter(
+                    ~Q(name=website.name), metadata__icontains=website.name
+                )
+                return Response(
+                    status=200,
+                    data={
+                        "Site dependencies": {
+                            "ocw www": WebsiteContentSerializer(
+                                instance=ocw_www_dependencies, many=True
+                            ).data,
+                            "course": WebsiteBasicSerializer(
+                                instance=course_dependencies, many=True
+                            ).data,
+                            "course content": WebsiteContentSerializer(
```

Comment:
Any reason for whitespace in the response property names? Could we use `ocw_www`, etc?

Reply:
@ChristopherChudzicki Sure.
can you have another look?